add Test for API methid: Create, Authorization and Update User, Creat…

VladimirKroshka opened the pull request:
…e and Get Orders

add Allure report

VladimirKroshka committed:
add Test for API methid: Create, Authorization and Update User, Create and Get Orders
add Allure report

pomegranate282922 commented on src/test/java/OrderCreationTest.java:
```diff
@@ -0,0 +1,153 @@
+
+    @Before
+    public void setUp() {
+        RestAssured.baseURI = BASE_URL;
```
Base uri присваиваем один раз в проекте. Можно использовать request specification в RestAssured или класс предок 
Нужно исправить

VladimirKroshka replied:
Вынес в отдельный класс BaseTest

pomegranate282922 commented on src/test/java/OrderCreationTest.java:
```diff
@@ -0,0 +1,153 @@
+        // Генерируем уникальные данные перед тестом
+        user = new User();
+        user.setEmail("test-data" + generateRandomNumber(5) + "@yandex.ru");
+        user.setPassword(generateRandomNumber(10));
```
Отлично что есть рандомизация

pomegranate282922 commented on src/test/java/UserCreationTest.java:
```diff
@@ -0,0 +1,105 @@
+    }
+
+    @Step("Создание уникального пользователя")
+    public void createUniqueUser() {
```
Все вызовы ручек выносим из тестовых классов чтобы их можно было переиспользовать в разных тестовых классах 
Нужно исправить

VladimirKroshka replied:
Вынес из всех степов ручки в отдельные методы в utils

pomegranate282922 commented on src/test/java/UserCreationTest.java:
```diff
@@ -0,0 +1,105 @@
+    public void createUserWithoutRequiredField() {
+        given()
+                .contentType("application/json")
+                .body("{\"email\": \"" + user.getEmail() + "\", \"password\": \"" + user.getPassword() + "\"}")
```
Тело запроса передаем через мапу или объект 
Нужно исправить

VladimirKroshka replied:
Теперь юзаю hashmap)

pomegranate282922 commented on src/test/java/resources/POJO/OrderRequest.java:
```diff
@@ -0,0 +1,20 @@
+package resources.POJO;
```
Пакеты именуем строчными буквами 
Нужно исправить

VladimirKroshka replied:
Done

pomegranate282922 commented on src/test/java/resources/POJO/OrderRequest.java:
```diff
@@ -0,0 +1,20 @@
+
+    public void setIngredients(List<String> ingredients) {
+        this.ingredients = ingredients;
+    }
```
Можно использовать библиотеку lombok для уменьшения boilerplate кода 
Можно улучшить

VladimirKroshka replied:
заюзал lombok, вышло намного лаконичней

pomegranate282922 commented on src/test/java/utils/RandomGenerator.java:
```diff
@@ -0,0 +1,15 @@
+public class RandomGenerator {
+
+    public static String generateRandomNumber(int length) {
+        Random random = new Random();
```
Можно использовать библиотеку faker 
Можно улучшить

VladimirKroshka replied:
Попробовал библиотеку faker, но надо разбираться (из коробки начал генерировать в password спецсимволы, сервер оказался к такому не готов = ) ) Оставил на потом разобраться)

pomegranate282922 reviewed (commented):
Работа хорошая! Все тесты реализованы. Есть рандомизация. Везде удаляются тестовые данные. Расставлены степы.
Замечания:
1. Base uri присваиваем один раз в проекте 
2. Тело запроса передаем через объект или мапу 
3. Нейминг пакетов 
4. Лучше попробовать избавиться от значительного дублирования кода и огромных раздутых тестовых классов из-за этого

VladimirKroshka committed:
added code based on review:
moved Base uri assignment to a separate method
request body is generated via hashmap
fixed package names
improved code duplication, moved API call out of tests

pomegranate282922 reviewed (approved):
Вот! Другое дело) отлично!
